cancer_ai/validator/rewarder.py

```diff
@@ -130,6 +130,14 @@ def get_top_hotkey(self, competition_id: str) -> Hotkey:
         # Update the current top hotkey and score
         self.current_top_hotkeys[competition_id] = (new_top_hotkey, new_top_score)
         return new_top_hotkey
+        
+    def get_hotkeys_with_non_zero_scores(self, competition_id) -> list[str]:
+        if competition_id not in self.average_scores or not self.average_scores[competition_id]:
+            raise ValueError(f"No hotkeys to choose from for competition {competition_id}")
+        scores = self.average_scores[competition_id]
```

Comment:
We get the non_zero scores from the average scores meaning we will possibly give minimal score to model copiers which were not slashed 5 times in a row (last 5 scores)